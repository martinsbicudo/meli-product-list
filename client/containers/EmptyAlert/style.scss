.EmptyAlert {
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;

  &Image { opacity: .5; }
}
