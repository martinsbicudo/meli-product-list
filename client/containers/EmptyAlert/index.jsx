import React from 'react';
import { object } from 'prop-types';
import EmptyState from '@Components/EmptyState';
import searchCleanImage from '@Client/static/images/search-clean.png';
import searchCleanImage2x from '@Client/static/images/search-clean@2x.png';
import style from './style.scss';

const EmptyAlert = ({
  empty,
}) => (
  <div className={style.EmptyAlert}>
    <EmptyState
      image={(
        <img
          className={style.EmptyAlertImage}
          src={searchCleanImage}
          srcSet={`${searchCleanImage2x} 2x`}
          alt="Lista Vazia"
          title="Lista Vazia"
        />
      )}
      title={empty.title || ''}
      description={empty.description || ''}
    />
  </div>
);

EmptyAlert.propTypes = {
  empty: object.isRequired,
};

export default EmptyAlert;
