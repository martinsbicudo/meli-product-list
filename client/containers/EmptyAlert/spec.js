import EmptyAlert from '.';

describe('Containers - <EmptyAlert />', () => {
  const testMessageTitle = 'TEST_TITLE';
  const testMessageDescription = 'TEST_DESCRIPTION';
  const createWrapper = (props = {}) => render(
    <EmptyAlert
      empty={props}
    />,
  );

  test('Should finds Title', () => {
    const { getByText } = createWrapper({
      title: testMessageTitle,
    });

    expect(getByText(testMessageTitle)).toBeInTheDocument();
  });

  test('Should finds Description', () => {
    const { getByText } = createWrapper({
      description: testMessageDescription,
    });

    expect(getByText(testMessageDescription)).toBeInTheDocument();
  });

  test('Wrapper Snapshot', () => {
    const { asFragment } = createWrapper();

    expect(asFragment()).toMatchSnapshot();
  });
});
