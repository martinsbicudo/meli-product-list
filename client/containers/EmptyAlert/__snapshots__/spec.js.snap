// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers - <EmptyAlert /> Wrapper Snapshot 1`] = `
<DocumentFragment>
  <div
    class="EmptyAlert"
  >
    <div
      class="EmptyState"
    >
      <figure
        class="EmptyStateFigure"
      >
        <img
          alt="Lista Vazia"
          class="EmptyAlertImage"
          src="[object Object]"
          srcset="[object Object] 2x"
          title="Lista Vazia"
        />
      </figure>
      <h1
        class="EmptyStateTitle"
      />
      <p
        class="EmptyStateDescription"
      />
    </div>
  </div>
</DocumentFragment>
`;
